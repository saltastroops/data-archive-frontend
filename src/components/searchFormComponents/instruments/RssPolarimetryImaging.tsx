import * as React from "react";
import MultiSelectField, {
  AllOption
} from "../../basicComponents/MultiSelectField";

interface IRssPolarimetryImagingProps {
  rssPolarimetryImaging?: string[];
  onSelect: (value: any) => void;
}

// Rss grating
const RSS_POLARIMETRY = ["Linear", "Linear Hi", "Circular", "All Stokes"];

/**
 * A form for selecting Rss Polarimetry Imaging-related search parameters.
 */
const RssPolarimetryImaging = (props: IRssPolarimetryImagingProps) => {
  const { onSelect, rssPolarimetryImaging } = props;

  // Function for handling change events
  const select = (e: React.SyntheticEvent<HTMLSelectElement>) => {
    const values = e.currentTarget.selectedOptions;
    onSelect({
      rssPolarimetryImaging: Array.from(values).map((t: any) => {
        return t.text;
      })
    });
  };

  return (
    <div>
      <p>RSS Polarimetry Imaging</p>
      <MultiSelectField
        name={"rssPolarimetryImaging"}
        onChange={select}
        value={rssPolarimetryImaging || ["All"]}
      >
        <AllOption />
        {RSS_POLARIMETRY.sort().map(i => (
          <option key={i} value={i}>
            {i}
          </option>
        ))}
      </MultiSelectField>
    </div>
  );
};

export default RssPolarimetryImaging;
