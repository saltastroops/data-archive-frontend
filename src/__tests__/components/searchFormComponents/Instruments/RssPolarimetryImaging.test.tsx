import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import RssPolarimetryImaging from "../../../../components/searchFormComponents/instruments/RssPolarimetryImaging";

const onSelect = jest.fn() as any;
describe("Rss Polarimetry Imaging", () => {
  it("should be defined", () => {
    expect(
      mount(
        <RssPolarimetryImaging rssPolarimetryImaging={[]} onSelect={onSelect} />
      )
    ).toBeDefined();
  });

  it("should render correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(
        mount(
          <RssPolarimetryImaging
            rssPolarimetryImaging={[]}
            onSelect={onSelect}
          />
        )
      )
    ).toMatchSnapshot();
  });
});
