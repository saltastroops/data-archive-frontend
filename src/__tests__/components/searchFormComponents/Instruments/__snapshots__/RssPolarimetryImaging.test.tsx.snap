// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rss Polarimetry Imaging should render correctly 1`] = `
.c0 {
  width: 95%;
}

<RssPolarimetryModeSelector
  onSelect={[MockFunction]}
  rssPolarimetryModes={Array []}
>
  <div>
    <p>
      RSS polarimetry mode
    </p>
    <MultiSelectField
      name="rssPolarimetry"
      onChange={[Function]}
      value={Array []}
    >
      <styled.div
        className="control select is-multiple "
      >
        <StyledComponent
          className="control select is-multiple "
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "
  width: 95%;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "usesTheme": false,
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0 control select is-multiple "
          >
            <select
              multiple={true}
              name="rssPolarimetry"
              onChange={[Function]}
              size={6}
              style={
                Object {
                  "width": "216px",
                }
              }
              value={Array []}
            >
              <AllOption>
                <option
                  value="All"
                >
                  All
                </option>
              </AllOption>
              <option
                key="All Stokes"
                value="All Stokes"
              >
                All Stokes
              </option>
              <option
                key="Circular"
                value="Circular"
              >
                Circular
              </option>
              <option
                key="Linear"
                value="Linear"
              >
                Linear
              </option>
              <option
                key="Linear Hi"
                value="Linear Hi"
              >
                Linear Hi
              </option>
            </select>
          </div>
        </StyledComponent>
      </styled.div>
    </MultiSelectField>
  </div>
</RssPolarimetryModeSelector>
`;
